File: pinochlecore/src/test/scala/bboreel/games/card/TestUtils.scala
package bboreel.games.card

import java.util.UUID

import bboreel.games.card.domain.{Card, FaceValue, Suite}

import scala.util.Random

/**
  * @author Robb Lee (robbmlee@gmail.com).
  */
object TestUtils {
  final val MaxCardValue = 20
  final val random: Random = new Random()

  def randomString(): String = {
    UUID.randomUUID().toString.replaceAll("-", "")
  }

  def randomPositiveInteger(max: Int = Int.MaxValue, min: Int = 0): Int = {
    var result = min - 1
    while (result < min) {
      result = random.nextInt(max)
    }
    result
  }

  def randomSuite(): Suite.Value = {
    val values = Suite.values.toArray
    values(randomPositiveInteger(values.size))
  }

  def createVectorOfCardsForAllSuites(ordinalRange: Range): Vector[Card] = {
    Suite.values.toVector.flatMap((suite: Suite.Value) => ordinalRange.toVector.map(
      (ordinal: Int) => Card(suite, FaceValue(ordinal, ordinal.toString))))
  }
}

File: card-deck/src/test/java/org/eelbbor/carddeck/TestUtils.java
package org.eelbbor.carddeck;

import java.util.Random;
import java.util.UUID;

public class TestUtils {
    private static final Random random = new Random();

    private TestUtils() {
    }

    public static int randomInteger() {
        return random.nextInt();
    }

    public static int randomInteger(int max) {
        return random.nextInt(max);
    }

    public static <E extends Enum<E>> E randomEnum(Class<E> enumData) {
        E[] enumConstants = enumData.getEnumConstants();
        return enumConstants[random.nextInt(enumConstants.length)];
    }

    public static String randomString() {
        return UUID.randomUUID().toString().replaceAll("-", "");
    }
}
